File: Konfigurator-nestjs-Backend/konfigurator-nestjs-backend/src/dto/get-ConfigurationDataResponse.dto.ts
import { ApiProperty} from '@nestjs/swagger';
import { BarrierDesigns } from "src/entities/BarrierDesignsEntity";
import { CharacterDesigns } from "src/entities/CharacterDesignsEntity";
import { TargetDesigns } from "src/entities/TargetDesignsEntity";
import { TileDesigns } from "src/entities/TileDesignsEntity";
import { DesignsDto } from './Designs.dto';

export class GetConfigurationDataResponseDto {
    constructor(
        barrierDesignDtos: DesignsDto[],
        characterDesignDtos: DesignsDto[],
        targetDesignDtos: DesignsDto[],
        tileDesignDtos: DesignsDto[]
    ) {
        this.barrierDesignDtos = barrierDesignDtos;
        this.characterDesignDtos = characterDesignDtos;
        this.targetDesignDtos = targetDesignDtos;
        this.tileDesignDtos = tileDesignDtos;
    }
    
    @ApiProperty()
    barrierDesignDtos: DesignsDto[];

    @ApiProperty()
    characterDesignDtos: DesignsDto[];

    @ApiProperty()
    targetDesignDtos: DesignsDto[];

    @ApiProperty()
    tileDesignDtos: DesignsDto[];
}

File: Konfigurator-aspnetcore-Backend/Konfigurator-aspnetcore-Backend/Models/DTOs/GetConfigurationDataResponseDto.cs
using System;

namespace Konfigurator_aspnetcore_Backend.Models.DTOs
{
	public class GetConfigurationDataResponseDto
    {
        public GetConfigurationDataResponseDto(
            List<DesignDto> barrierDesignDtos,
            List<DesignDto> characterDesignDtos,
            List<DesignDto> targetDesignDtos,
            List<DesignDto> tileDesignDtos)
        {
            this.barrierDesignDtos = barrierDesignDtos;
            this.characterDesignDtos = characterDesignDtos;
            this.targetDesignDtos = targetDesignDtos;
            this.tileDesignDtos = tileDesignDtos;
        }

        public List<DesignDto> barrierDesignDtos { get; }
        public List<DesignDto> characterDesignDtos { get; }
        public List<DesignDto> targetDesignDtos { get; }
        public List<DesignDto> tileDesignDtos { get; }
    }
}

